src/http/client.rs

```diff
@@ -688,6 +688,81 @@ impl Http {
         from_value(value).map_err(From::from)
     }
 
+    /// Creates a sticker.
+    pub async fn create_sticker<'a, T>(
+        &self,
+        guild_id: u64,
+        map: &JsonMap,
+        file: T,
+    ) -> Result<Sticker>
+    where
+        T: Into<AttachmentType<'a>>,
+    {
+        let mut multipart = reqwest::multipart::Form::new();
+
+        match file.into() {
+            AttachmentType::Bytes {
+                data,
+                filename,
+            } => {
+                multipart =
+                    multipart.part("file", Part::bytes(data.into_owned()).file_name(filename));
+            },
+            AttachmentType::File {
+                file,
+                filename,
+            } => {
+                let mut buf = Vec::new();
+                file.try_clone().await?.read_to_end(&mut buf).await?;
+
+                multipart = multipart.part("file", Part::stream(buf).file_name(filename));
+            },
+            AttachmentType::Path(path) => {
+                let filename =
+                    path.file_name().map(|filename| filename.to_string_lossy().into_owned());
+                let mut file = File::open(path).await?;
+                let mut buf = vec![];
+                file.read_to_end(&mut buf).await?;
+
+                let part = match filename {
+                    Some(filename) => Part::bytes(buf).file_name(filename),
+                    None => Part::bytes(buf),
+                };
+
+                multipart = multipart.part("file", part);
+            },
+            AttachmentType::Image(url) => {
+                let url = Url::parse(url).map_err(|_| Error::Url(url.to_string()))?;
+                let filename = url
+                    .path_segments()
+                    .and_then(|segments| segments.last().map(ToString::to_string))
+                    .ok_or_else(|| Error::Url(url.to_string()))?;
+                let response = self.client.get(url).send().await?;
+                let mut bytes = response.bytes().await?;
+                let mut picture: Vec<u8> = vec![0; bytes.len()];
+                bytes.copy_to_slice(&mut picture[..]);
+                multipart =
+                    multipart.part("file", Part::bytes(picture).file_name(filename.to_string()));
+            },
+        }
+
+        multipart = multipart.text("payload_json", serde_json::to_string(&map)?);
```

Comment:
This code seems to be repeated a lot in the library

Reply:
True. I was thinking if I can get #1269 fixed and removing the duplicate code with that fix since it's related, so it will probably be in a separate PR.